<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>600</width>
    <height>600</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::ClickFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetDefaultConstraint</enum>
   </property>
   <item>
    <widget class="GLWidget" name="widget_canvas" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="5" column="0">
      <widget class="QLabel" name="label_eyeY">
       <property name="text">
        <string>Глаз Y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_lampZ">
       <property name="text">
        <string>Лампа Z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="16" column="1">
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="18" column="0">
      <widget class="QLabel" name="label_fps">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Кадры:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="4" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_eyeX">
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
     <item row="19" column="1">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="17" column="1">
      <widget class="QPushButton" name="pushButton_timer">
       <property name="text">
        <string>Запустить таймеры</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_lampX">
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="11" column="0">
      <widget class="QLabel" name="label_bias">
       <property name="text">
        <string>Bias</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_eyeZ">
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>3.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="11" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_bias">
       <property name="decimals">
        <number>3</number>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
       <property name="value">
        <double>0.005000000000000</double>
       </property>
      </widget>
     </item>
     <item row="13" column="1">
      <widget class="QComboBox" name="comboBox_texture">
       <item>
        <property name="text">
         <string>Каменная стена</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Дерево</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Камень</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Кирпич</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_eyeY">
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>-3.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_lampZ">
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>2.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_scale">
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>0.060000000000000</double>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_eyeX">
       <property name="text">
        <string>Глаз X</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_lampY">
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_lampY">
       <property name="text">
        <string>Лампа Y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="13" column="0">
      <widget class="QLabel" name="label_texture">
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Текстура</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="18" column="1">
      <widget class="QLabel" name="label_fps_value">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="12" column="1">
      <spacer name="verticalSpacer_4">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_lampX">
       <property name="text">
        <string>Лампа X</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_eyeZ">
       <property name="text">
        <string>Глаз Z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="label_mapping">
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Mapping </string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_steps">
       <property name="decimals">
        <number>0</number>
       </property>
       <property name="minimum">
        <double>10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>200.000000000000000</double>
       </property>
       <property name="value">
        <double>10.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QComboBox" name="comboBox_mapping">
       <item>
        <property name="text">
         <string>Parallax Mapping</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Steep Parallax Mapping</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Relief Mapping</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Parallax Occlusion Mapping</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="label_steps">
       <property name="text">
        <string>Steps</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="label_scale">
       <property name="text">
        <string>Scale</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
